pkg/providers/amifamily/ami.go

```diff
@@ -132,6 +132,38 @@ func (p *Provider) Get(ctx context.Context, nodeTemplate *v1alpha1.AWSNodeTempla
 	return amiIDs, nil
 }
 
+func (p *Provider) GetAMIWithRequirements(ctx context.Context, nodeTemplate *v1alpha1.AWSNodeTemplate, instanceTypes []*cloudprovider.InstanceType, amiFamily AMIFamily) (map[AMI]scheduling.Requirements, error) {
```

Comment:
Is there a way to condense this and the `Get` call so that we aren't duplicating logic in both places? i.e. you can associate the AMIs first and then use that to iterate through the instance types to check compatability